src/input.rs

```diff
@@ -267,7 +267,7 @@ impl<'a> InputReader<'a> {
         };
 
         if content_type == Some(ContentType::UTF_16LE) {
```

Comment:
Same issue apply to both UTF16-LE and UTF16-BE, maybe a branch for UTF16-BE is needed as well 😦